### ISSUE TITLE
Web sample? 

### ISSUE THREAD

**Issue by trungnt2910** [itsmevjnk/HRng#1]
Many people don't want to download huge .NET 6 binaries so an online sample would be nice.

It's possible with Blazor or Uno Platform (the latter is preferred as we can also make mobile apps at the same time) along with a CORS proxy server.

**Comment by itsmevjnk** [comment]
I'm currently looking to create a web-based frontend for HRng. However, I think hosting a publicly accessible instance of said frontend will come with a number of issues, most of which stem from the CORS proxy server:
 - How can the user trust the server, knowing that it has full access to their Facebook account (through the means of cookies)? And even then, the server may be abused maliciously (e.g. a security breach that causes the credentials of other users to be leaked, or an attempt at DoS attacking Facebook, see below).
 - While unproven, there is a possibility that Facebook may impose some sort of IP ban on the CORS server due to the high amounts of requests concentrated in just one address (probably through intense rate-limiting and/or banning accounts associated with the server on the grounds that they are involved in scraping Facebook). How can we mitigate that without affecting user experience?

These are some of the problems that I can currently think of when it comes to making a public web frontend. It's still a good idea anyway, and I'd be quite happy to see solutions to the current issues.

**Comment by trungnt2910** [comment]
It is possible to rate-limit the public sample server as well as the CORS proxy, and then guide the users to install a certain Tampermonkey script that handles the cross-origin requests.

Clients with the Tampermonkey script enabled will then be able to use the full featured software.

**Comment by itsmevjnk** [comment]
So you mean the user's device is their own CORS server? That seems to be a great idea, will surely check that out.

**Comment by trungnt2910** [comment]
Yes, Tampermonkey scripts run from the context of a Chrome extension, which can request resources from any origin.

See: https://www.tampermonkey.net/documentation.php#GM_xmlhttpRequest
